### PR TITLE
Improve the verbosity & stability of os.getSupportedVersions tests

### PR DESCRIPTION
**Pull request by thgreasi**
Without the `expectSorted` change the tests might fail on staging when the `versions` array contains items of the same normalized semver value, depending on the length of the array.
Found this edge case when sorting the versions from
https://api.balena-staging.com/device-types/v1/raspberry-pi/images/
in which case `"2.0.0+rev1"` & `"2.0.0.rev1"` (which have the same normalized semver value) swapped positions after resorting.
Demo: https://runkit.com/embed/690i3z8wz9q2

Resolves: #667
Change-type: patch
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

---
##### Contributor checklist

- [x] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: tests/integration/models/os.spec.coffee [hunk 1]
```diff
@@ -90,14 +90,31 @@ describe 'OS model', ->
...
-				sortedVersions = _.clone(osVersions.versions)
-				sortedVersions.sort(osVersionRCompare)
-				return osVersions and
-					osVersions.versions and osVersions.versions.length and
-					_.isEqual(osVersions.versions, sortedVersions)
-					osVersions.latest and osVersions.recommended and osVersions.default and
-					osVersions.default is osVersions.recommended
+				m.chai.expect(osVersions).to.be.an('object')
+				m.chai.expect(osVersions).to.have.property('versions').that.is.an('array')
+				m.chai.expect(osVersions.versions).to.be.an('array')
```
Comment (karaxuna) [inline tests/integration/models/os.spec.coffee]: Isn't this covered on previous line? `that.is.an('array')`

File: tests/integration/models/os.spec.coffee [hunk 2]
```diff
@@ -90,14 +90,31 @@ describe 'OS model', ->
...
+				m.chai.expect(osVersions.versions).to.not.have.lengthOf(0)
+
+				expectSorted(osVersions.versions, osVersionRCompare)
+
+				m.chai.expect(osVersions).to.have.property('latest').that.is.a('string')
+				m.chai.expect(osVersions).to.have.property('recommended').that.is.a('string')
+				m.chai.expect(osVersions).to.have.property('default').that.is.a('string')
+				m.chai.expect(osVersions.default).to.equal(osVersions.recommended)
```
Comment (karaxuna) [inline tests/integration/models/os.spec.coffee]: Indentation issue?

File: tests/integration/models/os.spec.coffee (hunk 1 above)
Comment (thgreasi) [reply tests/integration/models/os.spec.coffee]: @karaxuna yea that's true :+1: I will push an update asap

File: tests/integration/models/os.spec.coffee (hunk 2 above)
Comment (thgreasi) [reply tests/integration/models/os.spec.coffee]: @karaxuna I don't see one. Can you be a bit more specific?

Comment (karaxuna) [reply tests/integration/models/os.spec.coffee]: Oh sorry, screen size was the problem :)
